pages/course/[course]/assignment/[assignment]/submit/index.tsx

```diff
@@ -0,0 +1,104 @@
+import withProtected from "../../../../../../util/withProtected";
+import { queryParamToNumber } from "../../../../../../util/misc";
+import { GetServerSidePropsContext, NextPage } from "next";
+import Badge, { BadgeVariant } from "Components/Badge";
+import Sidebar from "Components/Sidebar";
+import { User, Assignment } from "types";
+import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs";
+import Upload from "../../../../../../Components/UploadBox";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useState } from "react";
+
+type Submission = {
+  id: string;
+  is_late: boolean;
+  score: number | null;
+  flags: string[] | null;
+  student: User;
+};
```

Comment:
this should probably go in `types` since it maps directly to a DB object.

Reply:
Would it be a good example of using an interface, there is already a submission type, this is a stripped down version of it